libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1577,9 +1577,13 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 		if (memberFunctionType->kind() == FunctionType::Kind::Internal)
 		{
 			auto const& functionDefinition = dynamic_cast<FunctionDefinition const&>(memberFunctionType->declaration());
-			define(IRVariable(_memberAccess).part("functionIdentifier")) << to_string(functionDefinition.id()) << "\n";
 			if (!_memberAccess.annotation().calledDirectly)
+			{
+				define(IRVariable(_memberAccess).part("functionIdentifier")) <<
```

Comment:
Could maybe create a convenience function to return the function id and add the function to the internal dispatch.